utils/azExtResourceType.d.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+/**
+ * Normalized type for Azure resources that uniquely identifies resource type for the purposes
+ * of the Azure extensions
+ */
+export declare enum AzExtResourceType {
+    AppServices = 'AppServices',
+    AzureCosmosDb = 'AzureCosmosDb',
+    ContainerApps = 'ContainerApps',
+    ContainerAppsEnvironment = 'ContainerAppsEnvironment',
+    FunctionApp = 'FunctionApp',
+    PostgresqlServersFlexible = 'PostgresqlServersFlexible',
+    PostgresqlServersStandard = 'PostgresqlServersStandard',
+    StaticWebApps = 'StaticWebApps',
+    StorageAccounts = 'StorageAccounts',
+    VirtualMachines = 'VirtualMachines',
+
+    // Below are not supported but have icons in the Resources extension

```

Comment:
https://github.com/microsoft/vscode-azureresourcegroups/issues/351#issuecomment-1199793191

Regarding this comment: Should we maybe have another section of RelatedResourceTypes that are directly related/created by our extension, even if we don't support resolving them into parent nodes?

Reply:
We can always make those changes later. Also I'd like to discuss Eric's feedback with Matt first.